nixos/modules/services/audio/mpd.nix

```diff
@@ -123,10 +133,24 @@ in {
 
   config = mkIf cfg.enable {
 
+    systemd.sockets.mpd = mkIf cfg.startWhenNeeded {
+      description = "Music Player Daemon Socket";
+      wantedBy = [ "sockets.target" ];
+      listenStreams = [
+        "/run/mpd/socket"
+        "${cfg.network.listenAddress}:${toString cfg.network.port}"
+      ];
```

Comment:
Just to be sure: Did you test that mpd pick up multiple sockets, when passed via socket activation?